dbms/src/Flash/Pipeline/PipelineBuilder.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Flash/Pipeline/Pipeline.h>
+
+#include <vector>
+
+namespace DB
+{
+class PipelineIdGenerator
+{
+public:
+    UInt32 nextID()
+    {
+        return current_id++;
+    }
+
+private:
+    UInt32 current_id = 0;
+};
+using PipelineIdGeneratorPtr = std::shared_ptr<PipelineIdGenerator>;
+
+// PipelineBuilder is used to build pipelines, and PipelineBreaker is used to build dependencies between pipelines in the process.
+class PipelineBuilder
+{
+public:
+    PipelineBuilder()
+        : id_generator(std::make_shared<PipelineIdGenerator>())
+        , pipeline_breaker(std::nullopt)
+    {
+        pipeline = std::make_shared<Pipeline>(id_generator->nextID());
+    }
+
+private:
+    struct PipelineBreaker
+    {
+        PipelineBreaker(const PipelinePtr & dependent_, const PhysicalPlanNodePtr & breaker_node_)
```

Comment:
I think the argument name is not intuitive, may be we can use prev and next to show the dependency between pipeline.

Reply:
/cc @zanmato1984 
what do you think.

Reply:
Agree that the names are not intuitive. Maybe call them `pipeline` and `breaker`: so that people know that this `PipelineBreaker` is breaking this `pipeline` on this `breaker` node?